### PR TITLE
Updates for Julia 0.6

### LINKED ISSUE #192: using Winston fails 0.3.2, OSX
error:

``` shell
julia> using Winston
ERROR: `round` has no method matching round(::Type{Uint32}, ::Float64)
 in colormap at /Users/el24293/.julia/Winston/src/plot.jl:283
 in colormap at /Users/el24293/.julia/Winston/src/plot.jl:287
 in colormap at /Users/el24293/.julia/Winston/src/plot.jl:286
 in include at /Applications/Julia-0.3.2.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.3.2.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include at /Applications/Julia-0.3.2.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.3.2.app/Contents/Resources/julia/lib/julia/sys.dylib
 in reload_path at loading.jl:152
 in _require at loading.jl:67
 in require at loading.jl:51
while loading /Users/el24293/.julia/Winston/src/plot.jl, in expression starting on line 292
while loading /Users/el24293/.julia/Winston/src/Winston.jl, in expression starting on line 2614
```

version info:

``` shell
julia> versioninfo()
Julia Version 0.3.2
Commit 21d5433* (2014-10-21 20:18 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.3.0)
  CPU: Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

installed from binary. 
no issue using Color, Cairo, or Tk.

### LINKED ISSUE #204: [PkgEval] Winston may have a testing issue on Julia 0.3 (2015-02-13)
[PackageEvaluator.jl](https://github.com/IainNZ/PackageEvaluator.jl) is a script that runs nightly. It attempts to load all Julia packages and run their tests (if available) on both the stable version of Julia (0.3) and the nightly build of the unstable version (0.4). The results of this script are used to generate a [package listing](http://pkg.julialang.org/) enhanced with testing results.
#### On Julia 0.3
- On **2015-02-10** the testing status was `Tests pass.`
- On **2015-02-13** the testing status changed to `Tests fail, but package loads.`

`Tests pass.` means that PackageEvaluator found the tests for your package, executed them, and they all passed.

`Tests fail, but package loads.` means that PackageEvaluator found the tests for your package, executed them, and they didn't pass. However, trying to load your package with `using` worked.

_This issue was filed because your testing status became worse. No additional issues will be filed if your package remains in this state, and no issue will be filed if it improves. If you'd like to opt-out of these status-change messages, reply to this message saying you'd like to and @IainNZ will add an exception. If you'd like to discuss [PackageEvaluator.jl](https://github.com/IainNZ/PackageEvaluator.jl) please file an issue at the repository. For example, your package may be untestable on the test machine due to a dependency - an exception can be added._

_Test log_:

```
>>> 'Pkg.add("Winston")' log
INFO: Installing Cairo v0.2.24
INFO: Installing Color v0.4.0
INFO: Installing FixedPointNumbers v0.0.6
INFO: Installing Graphics v0.1.0
INFO: Installing IniFile v0.2.4
INFO: Installing Tk v0.2.17
INFO: Installing Winston v0.11.7
INFO: Building Cairo
INFO: Building Tk
INFO: Package database updated
INFO: METADATA is out-of-date a you may not have the latest version of Winston
INFO: Use `Pkg.update()` to get the latest versions of your packages

>>> 'using Winston' log
Julia Version 0.3.5
Commit a05f87b (2015-01-08 22:33 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

>>> test log

ERROR: `getgc` has no method matching getgc(::Canvas)
 in anonymous at /home/idunning/pkgtest/.julia/v0.3/Winston/src/tk.jl:23
 in display at /home/idunning/pkgtest/.julia/v0.3/Winston/src/tk.jl:35
 in include at ./boot.jl:245
 in include_from_node1 at ./loading.jl:128
 in include at ./boot.jl:245
 in include_from_node1 at loading.jl:128
 in process_options at ./client.jl:285
 in _start at ./client.jl:354
while loading /home/idunning/pkgtest/.julia/v0.3/Winston/test/canvas.jl, in expression starting on line 20
while loading /home/idunning/pkgtest/.julia/v0.3/Winston/test/runtests.jl, in expression starting on line 1
INFO: Testing Winston
===============================[ ERROR: Winston ]===============================

failed process: Process(`/home/idunning/julia03/usr/bin/julia /home/idunning/pkgtest/.julia/v0.3/Winston/test/runtests.jl`, ProcessExited(1)) [1]

================================================================================
INFO: No packages to install, update or remove
ERROR: Winston had test errors
 in error at error.jl:21
 in test at pkg/entry.jl:718
 in anonymous at pkg/dir.jl:28
 in cd at ./file.jl:20
 in cd at pkg/dir.jl:28
 in test at pkg.jl:67
 in process_options at ./client.jl:213
 in _start at ./client.jl:354


>>> end of log
```

### LINKED ISSUE #205: Can`t run Winston ERROR: LoadError: UndefVarError: Graphics not defined
Cant run Winston ERROR: LoadError: UndefVarError: Graphics not defined
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.0-dev+3600 (2015-02-25 15:26 UTC)
 _/ |\__'_|_|_|__'_|  |  Commit f96c23c\* (0 days old master)
|__/                   |  x86_64-w64-mingw32

julia> Pkg.status()
INFO: Initializing package repository C:\Users\SAMSUNG2.julia\v0.4
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
No packages installed

julia> Pkg.add("Winston")
INFO: Cloning cache of BinDeps from git://github.com/JuliaLang/BinDeps.jl.git
INFO: Cloning cache of Cairo from git://github.com/JuliaLang/Cairo.jl.git
INFO: Cloning cache of Color from git://github.com/JuliaLang/Color.jl.git
INFO: Cloning cache of Compat from git://github.com/JuliaLang/Compat.jl.git
INFO: Cloning cache of FixedPointNumbers from git://github.com/JeffBezanson/FixedPointNumbers.jl.git
INFO: Cloning cache of Graphics from git://github.com/JuliaLang/Graphics.jl.git
INFO: Cloning cache of IniFile from git://github.com/JuliaLang/IniFile.jl.git
INFO: Cloning cache of LibExpat from git://github.com/amitmurthy/LibExpat.jl.git
INFO: Cloning cache of SHA from git://github.com/staticfloat/SHA.jl.git
INFO: Cloning cache of Tk from git://github.com/JuliaLang/Tk.jl.git
INFO: Cloning cache of URIParser from git://github.com/JuliaWeb/URIParser.jl.git
INFO: Cloning cache of WinRPM from git://github.com/JuliaLang/WinRPM.jl.git
INFO: Cloning cache of Winston from git://github.com/nolta/Winston.jl.git
INFO: Cloning cache of Zlib from git://github.com/dcjones/Zlib.jl.git
INFO: Installing BinDeps v0.3.7
INFO: Installing Cairo v0.2.24
INFO: Installing Color v0.4.1
INFO: Installing Compat v0.2.12
INFO: Installing FixedPointNumbers v0.0.6
INFO: Installing Graphics v0.1.0
INFO: Installing IniFile v0.2.4
INFO: Installing LibExpat v0.0.6
INFO: Installing SHA v0.0.3
INFO: Installing Tk v0.2.17
INFO: Installing URIParser v0.0.4
INFO: Installing WinRPM v0.1.6
INFO: Installing Winston v0.11.7
INFO: Installing Zlib v0.1.7
INFO: Building WinRPM
WARNING: skipping repodata/repomd.xml, not in cache -- call WinRPM.update() to download
WARNING: beginswith is deprecated, use startswith instead.
 in depwarn at deprecated.jl:40
 in beginswith at deprecated.jl:10
 in anonymous at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:80
 in open at iostream.jl:137
 in getcachedir at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:76
 in update at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:131
 in **init** at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:40
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in reload_path at loading.jl:152
 in _require at loading.jl:67
 in require at loading.jl:51
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in evalfile at loading.jl:168 (repeats 2 times)
 in anonymous at pkg/entry.jl:624
 in cd at file.jl:30
 in build! at pkg/entry.jl:623
 in build! at pkg/entry.jl:619
 in build at pkg/entry.jl:635
 in resolve at pkg/entry.jl:446
 in edit at pkg/entry.jl:24
 in anonymous at task.jl:344
WARNING: split(x,y,l::Integer) is deprecated, use split(x,y; limit=l) instead.
 in depwarn at deprecated.jl:40
 in split at deprecated.jl:29
 in anonymous at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:81
 in open at iostream.jl:137
 in getcachedir at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:76
 in update at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:131
 in __init__ at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:40
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in reload_path at loading.jl:152
 in _require at loading.jl:67
 in require at loading.jl:51
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in evalfile at loading.jl:168 (repeats 2 times)
 in anonymous at pkg/entry.jl:624
 in cd at file.jl:30
 in build! at pkg/entry.jl:623
 in build! at pkg/entry.jl:619
 in build at pkg/entry.jl:635
 in resolve at pkg/entry.jl:446
 in edit at pkg/entry.jl:24
 in anonymous at task.jl:344
WARNING: skipping repodata/repomd.xml, not in cache -- call WinRPM.update() to download
WARNING: beginswith is deprecated, use startswith instead.
 in depwarn at deprecated.jl:40
 in beginswith at deprecated.jl:10
 in anonymous at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:80
 in open at iostream.jl:137
 in getcachedir at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:76
 in update at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:131
 in update at C:\Users\SAMSUNG2.julia\v0.4\WinRPM\src\WinRPM.jl:125
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in evalfile at loading.jl:168 (repeats 2 times)
 in anonymous at pkg/entry.jl:624
 in cd at file.jl:30
 in build! at pkg/entry.jl:623
 in build! at pkg/entry.jl:619
 in build at pkg/entry.jl:635
 in resolve at pkg/entry.jl:446
 in edit at pkg/entry.jl:24
 in anonymous at task.jl:344
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1//repodata/repomd.x
ml
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1//repodata/3804dc2e
4b633f07095ee8c36ac288bfb544355b763fa8f1fe0254ec716e6f8b-primary.xml.gz
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1//repodata/repomd.x
ml
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1//repodata/2b51f3de
32249d7afb94b5da88e64e0ff0e3b92ff3f139c51356dd7d2b70f081-primary.xml.gz
INFO: Building Cairo
WARNING: src::AbstractCmd |> dest::AbstractCmd is deprecated, use pipe(src,dest) instead.
 in depwarn at deprecated.jl:40
 in |> at deprecated.jl:29
 in unpack_cmd at C:\Users\SAMSUNG2.julia\v0.4\BinDeps\src\BinDeps.jl:95
 in lower at C:\Users\SAMSUNG2.julia\v0.4\BinDeps\src\BinDeps.jl:236
 in generate_steps at C:\Users\SAMSUNG2.julia\v0.4\BinDeps\src\dependencies.jl:236
 in lower at C:\Users\SAMSUNG2.julia\v0.4\BinDeps\src\dependencies.jl:254
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in evalfile at loading.jl:168 (repeats 2 times)
 in anonymous at pkg/entry.jl:624
 in cd at file.jl:30
 in build! at pkg/entry.jl:623
 in build at pkg/entry.jl:635
 in resolve at pkg/entry.jl:446
 in edit at pkg/entry.jl:24
 in anonymous at task.jl:344
INFO: Updating WinRPM package list
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1//repodata/repomd.x
ml
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1//repodata/repomd.x
ml
INFO: Packages to install: libintl8, libglib-2_0-0, libffi6, glib2-lang, libgobject-2_0-0, glib2
INFO: Downloading: libintl8
INFO: Extracting: libintl8
INFO: Downloading: libglib-2_0-0
INFO: Extracting: libglib-2_0-0
INFO: Downloading: libffi6
INFO: Extracting: libffi6
INFO: Downloading: glib2-lang
INFO: Extracting: glib2-lang
INFO: Downloading: libgobject-2_0-0
INFO: Extracting: libgobject-2_0-0
INFO: Downloading: glib2
INFO: Extracting: glib2
INFO: Complete
INFO: Packages to install: libxml2-2, libfreetype6, libpixman-1-0, zlib1, libgcc_s_seh1, fontconfig, libpng16-16,
libharfbuzz0, libcairo2
INFO: Downloading: libxml2-2
INFO: Extracting: libxml2-2
INFO: Downloading: libfreetype6
INFO: Extracting: libfreetype6
INFO: Downloading: libpixman-1-0
INFO: Extracting: libpixman-1-0
INFO: Downloading: zlib1
INFO: Extracting: zlib1
INFO: Downloading: libgcc_s_seh1
INFO: Extracting: libgcc_s_seh1
INFO: Downloading: fontconfig
INFO: Extracting: fontconfig
INFO: Downloading: libpng16-16
INFO: Extracting: libpng16-16
INFO: Downloading: libharfbuzz0
INFO: Extracting: libharfbuzz0
INFO: Downloading: libcairo2
INFO: Extracting: libcairo2
INFO: Complete
INFO: Packages to install: libgmodule-2_0-0, libpango-1_0-0
INFO: Downloading: libgmodule-2_0-0
INFO: Extracting: libgmodule-2_0-0
INFO: Downloading: libpango-1_0-0
INFO: Extracting: libpango-1_0-0
INFO: Complete
INFO: Building Tk

WARNING: deprecated syntax "[a=>b, ...]" at C:\Users\SAMSUNG2.julia\v0.4\Tk\deps\build.jl:54.
Use "Dict(a=>b, ...)" instead.
INFO: Packages to install: tcl
INFO: Downloading: tcl
INFO: Extracting: tcl
INFO: Complete
INFO: Packages to install: tk
INFO: Downloading: tk
INFO: Extracting: tk
INFO: Complete
INFO: Package database updated

julia> using Winston
ERROR: LoadError: UndefVarError: Graphics not defined
 in include at boot.jl:249
 in include_from_node1 at loading.jl:128
 in reload_path at loading.jl:152
 in _require at loading.jl:67
 in require at loading.jl:51
while loading C:\Users\SAMSUNG2.julia\v0.4\Winston\src\Winston.jl, in expression starting on line 5

julia> WinRPM.update()
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1//repodata/repomd.x
ml
INFO: Downloading http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1//repodata/repomd.x
ml

julia> using Winston

julia> plot(rand(10))
ERROR: UndefVarError: plot not defined

julia> In file  :Winston.jl in first lines is :

module Winston

using Cairo
using Color
importall Base.Graphics
using IniFile

export
    closefig,
    colormap,
    errorbar,
...
The dir C:\Users\SAMSUNG2.julia\v0.4 WAS cleanig befor installing.
Winston on this machine works 2 weekes ago :/

Paul

### LINKED ISSUE #241: Dates dependency
I just reinstalled Winston today. This went fine but when I tried `using Winston`, I got an error complaining that the `Dates` package is not installed. Installing this package solved the problem. I use julia 0.4rc.

### LINKED ISSUE #246: Unnecessary (?) parentheses raise WARNING
With julia 0.4.0, this warning is raised when using Winston:

```
WARNING: deprecated syntax "AbstractVecOrMat{T} (" at /?/?/.julia/v0.4/Winston/src/Winston.jl:96.
Use "AbstractVecOrMat{T}(" instead.
```

The code in question is:

```
if VERSION < v"0.3-"
    typealias AbstractVecOrMat{T} (@compat Union{AbstractVector{T}, AbstractMatrix{T}})
```

Replacing it with:

```
    typealias AbstractVecOrMat{T} @compat Union{AbstractVector{T}, AbstractMatrix{T}}
```

i.e. withouth parentheses removes the warning, but I'm not sure to which point this is relevant for julia users in versions < v0.3- (does this still even work?).

### PR DESCRIPTION
**Pull request by ararslan**
This PR improves support for Julia 0.6, drops support for 0.4 for ease of maintainability, and removes vestiges of old and unnecessary code.

Fixes #246
Fixes #241
Fixes #205
Fixes #204
Fixes #192

### CODE REVIEW

**Comment by ararslan** [comment]
There's more updating to be done (hence WIP), but currently the tests are failing on 0.6 because Tk, one of the dependencies, does not yet support 0.6.

File: REQUIRE [hunk 1]
```diff
@@ -1,8 +1,8 @@
-julia 0.4
+julia 0.5
 Cairo
 Colors
-IniFile
-Tk
-Compat 0.7.16
+IniFile 0.3.1
+Tk 0.5.1
```
Comment (tkelman) [inline REQUIRE]: earlier versions would work on 0.5, right? the min versions should be the widest over the supported julia versions - the broken deps' earlier versions could be julia upper bounded for 0.6 if necessary

Comment (ararslan) [inline REQUIRE]: Unclear, since Tk wasn't being tested on macOS at all, so it may well have been failing there on 0.5 as well.

**Comment by ararslan** [comment]
Here's something weird: If you run the following on Julia 0.5, you get a plot, and if you run it on 0.6, you get a message saying, "Winston: no data in range."
```julia
p = FramedPlot()
x = linspace(0, 3π, 100)
add(p, Curve(x, cos.(x)))
```
The same is true even for simpler `x`, e.g. `x = [1, 2, 3, 4]`.

I've been banging my head against this but haven't been able to pinpoint why there would be a difference in behavior between 0.5 and 0.6. @timholy, do you by chance have any insight? (I ask you in particular because you know the subtleties between 0.5 and 0.6 well and are a top contributor here.)

**Comment by maxruby** [comment]
Description:  Request `ERROR: LoadError: UndefVarError: fp not defined` when trying to use Winston with v0.6

`System and Julia`:
MacOS Sierra 10.12.3
Julia Version: 0.6.0-pre.alpha.49
Commit eca966c* (2017-03-04 22:22 UTC)
LLVM: libLLVM-3.9.0 (ORCJIT, haswell)
Winston:  `master`,  commit `88e44c20b765edeff545ef3c4c11667a484401fe` 

`Error`:
```julia
julia> using Winston

WARNING: deprecated syntax "abstract HasAttr" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:118.
Use "abstract type HasAttr end" instead.

WARNING: deprecated syntax "abstract HasStyle<:HasAttr" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:119.
Use "abstract type HasStyle<:HasAttr end" instead.

WARNING: deprecated syntax "abstract PlotComponent<:HasStyle" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:120.
Use "abstract type PlotComponent<:HasStyle end" instead.

WARNING: deprecated syntax "abstract PlotContainer<:HasAttr" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:121.
Use "abstract type PlotContainer<:HasAttr end" instead.

WARNING: deprecated syntax "typealias PlotAttributes Associative" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:123.
Use "const PlotAttributes = Associative" instead.

WARNING: deprecated syntax "typealias Box Union{BoundingBox,Rectangle}" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:178.
Use "const Box = Union{BoundingBox,Rectangle}" instead.

WARNING: deprecated syntax "abstract ErrorBar<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:310.
Use "abstract type ErrorBar<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract _Inset<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:406.
Use "abstract type _Inset<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract HalfAxis<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:596.
Use "abstract type HalfAxis<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract LineComponent<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:1980.
Use "abstract type LineComponent<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract LabelComponent<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:2252.
Use "abstract type LabelComponent<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract FillComponent<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:2352.
Use "abstract type FillComponent<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract ImageComponent<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:2447.
Use "abstract type ImageComponent<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract FramedComponent<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:2481.
Use "abstract type FramedComponent<:PlotComponent end" instead.

WARNING: deprecated syntax "abstract SymbolDataComponent<:PlotComponent" at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:2546.
Use "abstract type SymbolDataComponent<:PlotComponent end" instead.

WARNING: deprecated syntax "typealias INIVAL Union{AbstractString,Number,Bool,Void}" at /Users/Max/.julia/v0.6/IniFile/src/IniFile.jl:26.
Use "const INIVAL = Union{AbstractString,Number,Bool,Void}" instead.

WARNING: deprecated syntax "typealias HTSS Dict{AbstractString,INIVAL}" at /Users/Max/.julia/v0.6/IniFile/src/IniFile.jl:27.
Use "const HTSS = Dict{AbstractString,INIVAL}" instead.

WARNING: deprecated syntax "abstract AbstractProjection1" at /Users/Max/.julia/v0.6/Winston/src/geom.jl:67.
Use "abstract type AbstractProjection1 end" instead.

WARNING: deprecated syntax "abstract AbstractProjection2" at /Users/Max/.julia/v0.6/Winston/src/geom.jl:68.
Use "abstract type AbstractProjection2 end" instead.

WARNING: deprecated syntax "abstract Renderer" at /Users/Max/.julia/v0.6/Winston/src/renderer.jl:35.
Use "abstract type Renderer end" instead.

WARNING: deprecated syntax "abstract AbstractPainter" at /Users/Max/.julia/v0.6/Winston/src/paint.jl:29.
Use "abstract type AbstractPainter end" instead.

WARNING: deprecated syntax "typealias PlotArg Union{AbstractString,AbstractVector,AbstractMatrix,Function,Real}" at /Users/Max/.julia/v0.6/Winston/src/plot.jl:108.
Use "const PlotArg = Union{AbstractString,AbstractVector,AbstractMatrix,Function,Real}" instead.

WARNING: deprecated syntax "typealias Interval @compat Tuple{Real,Real}" at /Users/Max/.julia/v0.6/Winston/src/plot.jl:251.
Use "const Interval = @compat Tuple{Real,Real}" instead.

WARNING: deprecated syntax "typealias ScatterPlotPoints{T<:Real,S<:Real} @compat Tuple{Vector{T},Vector{S}}" at /Users/Max/.julia/v0.6/Winston/src/plot_interfaces.jl:13.
Use "ScatterPlotPoints{T<:Real,S<:Real} = @compat Tuple{Vector{T},Vector{S}}" instead.

WARNING: deprecated syntax "typealias ParametricFunctionPair @compat Tuple{Function,Function}" at /Users/Max/.julia/v0.6/Winston/src/plot_interfaces.jl:25.
Use "const ParametricFunctionPair = @compat Tuple{Function,Function}" instead.
ERROR: LoadError: UndefVarError: fp not defined
Stacktrace:
 [1] setattr(::Winston.FramedPlot, ::Symbol, ::Float64) at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:1095
 [2] #iniattr#117(::Array{Any,1}, ::Function, ::Winston.FramedPlot) at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:2689
 [3] (::Winston.##call#67#68)(::Array{Any,1}, ::Type{T} where T) at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:1068
 [4] Winston.FramedPlot() at /Users/Max/.julia/v0.6/Winston/src/Winston.jl:1044
 [5] include_from_node1(::String) at ./loading.jl:539
 [6] eval(::Module, ::Any) at ./boot.jl:235
 [7] require(::Symbol) at ./loading.jl:453
while loading /Users/Max/.julia/v0.6/Winston/src/Winston.jl, in expression starting on line 2825
```

**Comment by ararslan** [comment]
Hi @maxruby. Issues should be posted to the [issue tracker](https://github.com/nolta/Winston.jl/issues) rather than on pull requests. The issues you're experiencing should eventually be fixed once I've finished this PR.

**Comment by maxruby** [comment]
No problem. I deleted the comment, and I will not file any issues. Thanks for your time.

**Comment by timholy** [comment]
julia-0.5:
```julia
julia> min(NaN, 0.0)
0.0
```

julia-0.6:
```julia
julia> min(NaN, 0.0)
NaN
```

Start poking around https://github.com/nolta/Winston.jl/blob/8b5e10e1a5916d603dd1c8e50ece9e2897d82a30/src/Winston.jl#L160-L173

**Comment by ararslan** [comment]
Thanks so much @timholy! I had no idea that `min(NaN, 0.0)` changed.

**Comment by ararslan** [comment]
Looks like there's more Tk fixing to be done...

**Pull request merged**